include/envoy/server/filter_config.h

```diff
@@ -201,6 +202,15 @@ class NamedHttpFilterConfigFactory : public ProtocolOptionsFactory {
    * @return bool true if this filter must be the last filter in a filter chain, false otherwise.
    */
   virtual bool isTerminalFilter() { return false; }
+
+  /**
+   * @return FilterDependencies specification of dependencies required or
+   * provided on the decode and encode paths.
+   */
+  virtual envoy::extensions::filters::common::dependency::v3::FilterDependencies dependencies() {
```

Comment:
Nit: probably `const&`

Reply:
Oh, I see what's going on. This is more similar I think to `createEmptyRouteConfigProto()` above. I think you should return a `unique_ptr` with the allocated dependencies in this case.

Reply:
I don't understand what the value of that would be; the dependency specification protos won't typically be very large, so passing it by copy seems ok to me. What do you think? 

@snowp in case you have any thoughts

Reply:
Right, the protos contain multiple strings and we try not copy them around in general.